File: 2024. Maximize the Confusion of an Exam/Solution.cpp
class Solution {
public:
    int maxConsecutiveAnswers(string answerKey, int k) {
        int n = answerKey.size(), left = 0, ts = 0, fs = 0, count = 0, high = 0;
        for (int right = 0; right < n; right++) {
            if (answerKey[right] == 'T') {
                ts++;
            } else {
                fs++;
            }
            count = min(ts, fs);
            while (count > k) {
                if (answerKey[left] == 'T') {
                    ts--;
                } else {
                    fs--;
                }
                count = min(ts, fs);
                left++;
            }
            high = max(high, right - left + 1);
        }
        return high;
    }
};

File: 2024. Maximize the Confusion of an Exam/Solution.py
class Solution:
    def maxConsecutiveAnswers(self, answerKey: str, k: int) -> int:
        n, left, high, ts, fs, count = len(answerKey), 0, 0, 0, 0, 0
        for right in range(n):
            if answerKey[right] == 'T':
                ts += 1
            else:
                fs += 1
            count = min(ts, fs)
            while count > k:
                if answerKey[left] == 'T':
                    ts -= 1
                else:
                    fs -= 1
                count = min(ts, fs)
                left += 1
            high = max(high, right - left + 1)
        return high
